src/wp-includes/blocks.php

```diff
@@ -1002,6 +1002,123 @@ function set_ignored_hooked_blocks_metadata( &$parsed_anchor_block, $relative_po
+ *                                                       the markup for a given list of blocks that are hooked to it.
+ *                                                       Default: 'insert_hooked_blocks'.
+ * @return string The serialized markup.
+ */
+function apply_block_hooks_to_content( $content, $context, $callback = 'insert_hooked_blocks' ) {
+	$hooked_blocks = get_hooked_blocks();
+	if ( empty( $hooked_blocks ) && ! has_filter( 'hooked_block_types' ) ) {
+		return $content;
+	}
+
+	$blocks = parse_blocks( $content );
+
+	$before_block_visitor = make_before_block_visitor( $hooked_blocks, $context, $callback );
+	$after_block_visitor  = make_after_block_visitor( $hooked_blocks, $context, $callback );
+
+	return traverse_and_serialize_blocks( $blocks, $before_block_visitor, $after_block_visitor );
+}
+
+/**
+ * Accepts the serialized markup of a block and its inner blocks, and returns serialized markup of the inner blocks.
+ *
+ * @since 6.6.0
+ * @access private
+ *
+ * @param string $serialized_block The serialized markup of a block and its inner blocks.
+ * @return string
+ */
+function remove_serialized_parent_block( $serialized_block ) {
+	$start = strpos( $serialized_block, '-->' ) + strlen( '-->' );
+	$end   = strrpos( $serialized_block, '<!--' );
+	return substr( $serialized_block, $start, $end - $start );
+}
+
+/**
+ * Updates the wp_postmeta with the list of ignored hooked blocks where the inner blocks are stored as post content.
+ * Currently only supports `wp_navigation` post types.
+ *
+ * @since 6.6.0
+ * @access private
+ *
+ * @param stdClass $post Post object.
+ * @return stdClass The updated post object.
+ */
+function update_ignored_hooked_blocks_postmeta( $post ) {
+	/*
+	 * In this scenario the user has likely tried to create a navigation via the REST API.
+	 * In which case we won't have a post ID to work with and store meta against.
+	 */
+	if ( empty( $post->ID ) ) {
+		return $post;
+	}
+
+	/**
+	 * Skip meta generation when consumers intentionally update specific Navigation fields
+	 * and omit the content update.
+	 */
+	if ( ! isset( $post->post_content ) ) {
+		return $post;
+	}
+
+	/**
+	 * Skip meta generation when the post content is not a navigation block.
+	 */
+	if ( ! isset( $post->post_type ) || 'wp_navigation' !== $post->post_type ) {
+		return $post;
+	}
+
+	$attributes = array();
+
+	if ( isset( $post->ID ) ) {
+		$ignored_hooked_blocks = get_post_meta( $post->ID, '_wp_ignored_hooked_blocks', true );
+		if ( ! empty( $ignored_hooked_blocks ) ) {
+			$ignored_hooked_blocks  = json_decode( $ignored_hooked_blocks, true );
+			$attributes['metadata'] = array(
+				'ignoredHookedBlocks' => $ignored_hooked_blocks,
+			);
+		}
+	}
+
+	$markup = get_comment_delimited_block_content(
+		'core/navigation',
+		$attributes,
+		$post->post_content
+	);
+
+	$serialized_block = apply_block_hooks_to_content( $markup, get_post( $post->ID ), 'set_ignored_hooked_blocks_metadata' );
```

Comment:
The `$context` here will reflect what's currently in the DB, but we should be passing the new content that's coming in over the wire. This is a similar problem to [the one we had with templates](https://core.trac.wordpress.org/ticket/60754). I don't think we should do it as part of this PR but separately; I'll just leave some notes here for now.

The fix more straight-forward here, as we don't have to map a `$changes` `stdClass` instance to a `WP_Block_Template` type, but only to a `WP_Post` object. We'll still need to apply the `$changes` on top of the existing `wp_navigation` post object.

Something like the following (inspired by [this](https://github.com/WordPress/wordpress-develop/pull/6278/files#diff-1d8e05d3ea69557ca8742330e62c1893a72d4ca4ff2ed4ccc5b94bcf5d20fed7R1510-R1522) -- minus the revision handling, which we don't need):

```suggestion
	// Find the existing post object.
	$existing_post = get_post( $post->ID );

	// Apply the changes to the existing post object.
	$post = (object) array_merge( (array) $existing_post, (array) $post );

	$serialized_block = apply_block_hooks_to_content( $markup, $post, 'set_ignored_hooked_blocks_metadata' );
```